istioctl/cmd/istioctl/verifyinstall.go

```diff
@@ -0,0 +1,330 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	"go.uber.org/multierr"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/log"
+	v1batch "k8s.io/api/batch/v1"
+	"k8s.io/api/extensions/v1beta1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kclient "k8s.io/client-go/kubernetes"
+)
+
+var (
+	installFileName string
+	resourceList    []*unstructured.Unstructured
+	kubeClient      kclient.Interface
+	extensionClient *apiextensionsclient.Clientset
+)
+
+var (
+	verifyInstallCmd = &cobra.Command{
+		Use:   "verify-install",
+		Short: "Validate Istio Installation Status",
+		Long: `
+		verify-install Validate Istio Installation Status
+`,
+		Example: `
+istioctl verify-install -f istio-demo.yaml
+`,
+		RunE: func(c *cobra.Command, _ []string) (err error) {
+			if err = validateInputFlags(); err != nil {
+				return err
+			}
+
+			var reader io.Reader
+			if installFileName == "-" {
+				reader = os.Stdin
+			} else {
+				var in *os.File
+				if in, err = os.Open(installFileName); err != nil {
+					return err
+				}
+				reader = in
+				defer func() {
+					if errClose := in.Close(); errClose != nil {
+						log.Errorf("Error: close file from %s, %s", installFileName, errClose)
+						if err == nil {
+							err = errClose
+						}
+					}
+				}()
+			}
+			err = buildKubeClient(kubeconfig)
+			if err != nil {
+				return err
+			}
+			err = buildResourceListFromInstallationFile(reader)
+			if err != nil {
+				return err
+			}
+			return loopThroughResources(c.OutOrStdout())
+		},
+	}
+)
+
+func init() {
+	experimentalCmd.AddCommand(verifyInstallCmd)
+
+	verifyInstallCmd.PersistentFlags().StringVarP(&installFileName, "filename", "f",
+		"", "Input Istio Installation File, eg: istio-demo.yaml")
+}
+
+func buildResourceListFromInstallationFile(in io.Reader) error {
```

Comment:
Any chance you can split this into at least two files: one for the command-line binding, the other for the core functionality.

Reply:
ok